upgrade/scripts/upgrade/util/upgrade-cray-nls.sh

```diff
@@ -50,38 +50,57 @@ function patchKeycloak() {
-    webOrigins=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .webOrigins')
 
-    id=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .id')
+    res_file=$(mktemp)
+    http_code=$(curl -s -o "${res_file}" -w "%{http_code}" -k -XGET -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients")
+    if [[ ${http_code} -ne 200 ]]; then
+        echo "Request Failed, Response code: ${http_code}"
+        cat "${res_file}"
+        exit 1
+    fi
+    
+    id=$(jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .id'  < "${res_file}" )
+    echo "Get client id: ${id}"
 
-    curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients/${id}" > client.json
-    if [[ ${foundArgoInRedirectUris} -ge 1 ]];then
-        echo "Argo URL is set in RedirectUris"
-        echo "${redirectUris}"
+    http_code=$(curl -s -o "client.json" -w "%{http_code}" -k -XGET -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients/${id}")
+    if [[ ${http_code} -ne 200 ]]; then
+        echo "Request Failed, Response code: ${http_code}"
+        cat "${res_file}"
+        exit 1
     else
-        echo
-        cat client.json | jq ".redirectUris[.redirectUris | length] |= . + \"https://$argoUrl/oauth/callback\""  | tee client.json
+        echo "Get client json"
     fi
-
-    if [[ ${foundArgoInWebOrigins} -ge 1 ]];then
-        echo "Argo URL is set in WebOrigins"
-        echo "${webOrigins}"
+    
+    echo "Configure RedirectUris"
+    if jq '.[] | select(.clientId=="oauth2-proxy-customer-management") | .redirectUris' < "${res_file}" | grep "https://argo.cmn"; then
+        echo "  Argo URL is set in RedirectUris"
     else
-        echo
-        #shellcheck disable=SC2002
-        cat client.json | jq ".webOrigins[.webOrigins | length] |= . + \"https://$argoUrl\"" | tee client.json
+        tmpFile=$(mktemp)
+        jq ".redirectUris[.redirectUris | length] |= . + \"https://$argoUrl/oauth/callback\"" < client.json  | tee "${tmpFile}" > /dev/null
+        mv "${tmpFile}" client.json
+    fi
+    
+    echo "Configure WebOrigins"
+    if jq '.[] | select(.clientId=="oauth2-proxy-customer-management") | .webOrigins'  < "${res_file}" | grep "https://argo.cmn"; then
+        echo "  Argo URL is set in WebOrigins"
+    else
+        tmpFile=$(mktemp)
+        jq ".webOrigins[.webOrigins | length] |= . + \"https://$argoUrl\"" < client.json  | tee "${tmpFile}"  > /dev/null
+        mv "${tmpFile}" client.json
     fi
 
-    curl -k -XPUT \
+    res_file=$(mktemp)
```

Comment:
You should clean up this tmp file.   I see 18000 temp files on fanta from the weekend testing.
```
ncn-m001:~ # ls /tmp/tmp.* | wc -l
18814
```

Reply:
throw out bash, problem solved